graphium/trainer/predictor.py

```diff
@@ -96,20 +95,46 @@ def __init__(
             )
             eval_options[task].check_metrics_validity()
 
+        # Work-around to retain task level when model_kwargs are modified for FullGraphFinetuningNetwork
+        if "task_heads_kwargs" in model_kwargs.keys():
+            task_heads_kwargs = model_kwargs["task_heads_kwargs"]
+        elif "pretrained_model_kwargs" in model_kwargs.keys():
+            # This covers finetuning cases where we finetune from the task_heads
+            task_heads_kwargs = model_kwargs["pretrained_model_kwargs"]["task_heads_kwargs"]
+        else:
+            raise ValueError("incorrect model_kwargs")
```

Comment:
I don't think this should be here. I think that, if you are using a pre-trained model, you should pass directly `model_kwargs["pretrained_model_kwargs"]` into `model_kwargs`.

Reply:
Removed in a3d471540bc3f69cc0af2346ab0e66141b65311b as explained below